packages/wallet/src/services/TransactionsTracker.ts

```diff
@@ -234,8 +236,14 @@ export const createTransactionsTracker = (
   const transactionsSource$ = new TrackerSubject(txSource$);
 
   const historicalTransactions$ = createHistoricalTransactionsTrackerSubject(transactionsSource$);
+
+  const [onChainNewTxPhase2Failed$, onChainNewTxSuccess$] = partition(
+    newTransactions$(historicalTransactions$).pipe(share()),
+    (tx) => Cardano.util.isPhase2ValidationErrTx(tx)
+  );
+
   const txConfirmed$ = (evt: OutgoingTx): Observable<ConfirmedTx> =>
-    newTransactions$(historicalTransactions$).pipe(
```

Comment:
Needs attention.
Previously this was recreating the newTransactions$ observable based on `historicalTransactions$` every time txConfirmed$ was called.
Now it reuses the `onChainNewTxSuccess$` observable.
Tests are passing but a vigilant pair of eyes are welcome.

Reply:
Looks good to me